src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -1872,6 +1872,11 @@ private void doGenSecretKey(String alias, String keyAlgName,
             System.err.println(form.format(source));
         }
 
+        CertPathConstraintsParameters cpcp =
+                new CertPathConstraintsParameters(secKey, null, null, null);
```

Comment:
Using `CertPathConstraintsParameters` is a little odd here, even though it works. I suggest creating a local `SecretKeyConstraintsParameters` subclass with a ctor with just the parameters you need and overriding the methods as needed. And changing `checkWeakConstraints` to take a `ConstraintsParameters` instead of `CertPathConstraintsParameters`.